pkg/networkservice/common/connect/client.go

```diff
@@ -34,7 +34,11 @@ type connectClient struct {
 }
 
 func (c *connectClient) Request(ctx context.Context, request *networkservice.NetworkServiceRequest, opts ...grpc.CallOption) (*networkservice.Connection, error) {
-	return next.Client(ctx).Request(ctx, request, opts...)
+	resp, err := next.Client(ctx).Request(ctx, request, opts...)
+	if err != nil {
```

Comment:
It doesn't mean that there is something wrong with server, it can be something wrong with request.

Reply:
Or it can be something wrong with someone (not next host) else in chain.

Reply:
Yep, it is also is covered by TestNSMGR_SelectsRestartingEndpoint :)  
Fixed.